dz13/src/app.js

```diff
@@ -0,0 +1,121 @@
+function isSomeProductInStock(productsList) {
+  return productsList.some((product) => product.inStock)
+}
+
+function makeUserTotalBalance(transactions) {
+  const newTransactions = transactions.reduce((result, transaction) => {
+    if (transaction.type === 'income') {
+      return result + transaction.amount
+    }
+    return result - transaction.amount
+  }, 0)
+  return newTransactions
+}
+
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+function getObjectKeysOrValues(object, type) {
+  return type === 'key' ? Object.keys(object) : Object.values(object)
+}
+
+function logInCheck(carts) {
+  Object.entries(carts).forEach((cart) => (console.log(`${cart[0]}: ${cart[1]} шт.`)))
+}
+
+function mergeSettings(defaultSettings, userPreferences) {
+  return { ...defaultSettings, ...userPreferences }
+}
+
+function checkUserAge(user) {
+  try {
+    if (user.age < 18) {
+      throw new Error('Меньше 18 лет')
+    }
+    console.log('Больше 18 лет(')
+  } catch (error) {
+    if (error.message === 'Меньше 18 лет') {
+      console.log('Меньше 18 лет(')
+    }
+  }
+}
+
+function checkNumber(number) {
+  return new Promise((resolve, reject) => {
+    if (number > 10) {
+      resolve('Число больше 10')
+    } else {
+      reject(new Error('Число меньше или равно 10'))
+    }
+  })
+}
+
+function delayedMessage(string, time) {
+  setTimeout(() => {
+    console.log(string)
+  }, time)
+}
+
+async function fetchProducts() {
+  try {
+    const response = await fetch('https://dummyjson.com/products?limit=10&select=title,price')
+    if (!response.ok) {
+      throw new Error('Не найден json')
```

Comment:
```suggestion
      throw new Error('Ошибка при выполнении запроса')
```

здесь именно ошибка при выполнении запроса

`ok` будет при успешном запросе и не `ok` в остальных случаях, например когда интернет отключился - в таком случае "не найден json" будет ложным сообщением